src/mongoc/mongoc-topology-scanner.c

```diff
@@ -47,11 +48,31 @@ mongoc_topology_scanner_new (mongoc_topology_scanner_cb_t cb,
    BSON_APPEND_INT32 (&ts->ismaster_cmd, "isMaster", 1);
 
    ts->cb = cb;
-   ts->cb_data = cb_data;
+   ts->cb_data = data;
+   ts->uri = uri;
```

Comment:
topology_new copies the uri, and cleans it up.  so the scanner can have a pointer and not worry about ownership